File: cryptography/src/main/java/com/personal/image_toolkit/TrustedCertificateEntryDataInterface.java
package com.personal.image_toolkit;

import java.security.cert.Certificate;

/**
 * Defines the contract for accessing data related to a Trusted Certificate Entry.
 */
public interface TrustedCertificateEntryDataInterface {

    /**
     * @return The alias of the certificate entry.
     */
    String getAlias();

    /**
     * @return The trusted certificate.
     */
    Certificate getCertificate();
}

File: cryptography/src/main/kotlin/com/personal/image_toolkit/TrustedCertificateEntryDataInterface.kt
package com.personal.image_toolkit

import java.security.cert.Certificate

/**
 * Defines the contract for accessing data related to a Trusted Certificate Entry.
 */
interface TrustedCertificateEntryDataInterface {
    /**
     * @return The alias of the certificate entry.
     */
    val alias: String

    /**
     * @return The trusted certificate.
     */
    val certificate: Certificate
}